front-end/src/screens/NotificationSettings/index.tsx

```diff
@@ -0,0 +1,180 @@
+	const [changed, setChanged] = useState<boolean>(false);
+	const [postParticipated, setPostParticipated] = useState<boolean>(false);
+	const [postCreated, setPostCreated] = useState<boolean>(false);
+	const [newProposal, setNewProposal] = useState<boolean>(false);
+	const [ownProposal, setOwnProposal] = useState<boolean>(false);
+	const [changeNotificationPreferenceMutation, { error }] = useChangeNotificationPreferenceMutation();
+	const { queueNotification } = useContext(NotificationContext);
+
+	useEffect(() => {
+		setPostParticipated(notification?.postParticipated || false);
+		setPostCreated(notification?.postCreated || false);
+		setNewProposal(notification?.newProposal || false);
+		setOwnProposal(notification?.ownProposal || false);
+	}, [notification]);
+
+	const updatePreference = () => {
+		changeNotificationPreferenceMutation({
+			variables: {
+				newProposal,
+				ownProposal,
+				postCreated,
+				postParticipated
+			}
+		})
+			.then(({ data }) => {
+				if (data && data.changeNotificationPreference && data.changeNotificationPreference.message) {
+					queueNotification({
+						header: 'Success!',
+						message: data.changeNotificationPreference.message,
+						status: NotificationStatus.SUCCESS
+					});
+					setChanged(false);
+				}
+				if (data && data.changeNotificationPreference && data.changeNotificationPreference.token) {
+					handleTokenChange(data.changeNotificationPreference.token);
+				}
+			}).catch((e) => {
+				queueNotification({
+					header: 'Failed!',
+					message: e.message,
+					status: NotificationStatus.ERROR
+				});
+				console.error('change name error', e);
+			});
+	};
+
+	const handlePostParticipatedChange = (event: React.FormEvent<HTMLInputElement>, data: CheckboxProps) => {
+		setPostParticipated(data.checked || false);
+		setChanged(true);
+	};
+
+	const handlePostCreatedChange = (event: React.FormEvent<HTMLInputElement>, data: CheckboxProps) => {
+		setPostCreated(data.checked || false);
+		setChanged(true);
+	};
+
+	const handleNewProposalChange = (event: React.FormEvent<HTMLInputElement>, data: CheckboxProps) => {
+		setNewProposal(data.checked || false);
+		setChanged(true);
+	};
+
+	const handleOwnProposalChange = (event: React.FormEvent<HTMLInputElement>, data: CheckboxProps) => {
+		setOwnProposal(data.checked || false);
+		setChanged(true);
+	};
+
+	return (
+		<Grid>
+			<Grid.Column className={className} mobile={16} tablet={12} computer={12} largeScreen={10} widescreen={10}>
+				<h2>Notification Preferences</h2>
+				<Form standalone={false}>
+					<Form.Group>
+						<Form.Field>
+							<Checkbox label='Subscribe to post you participate in' checked={postParticipated} toggle onChange={handlePostParticipatedChange} />
+						</Form.Field>
+					</Form.Group>
+					<Form.Group>
+						<Form.Field>
+							<Checkbox label='Subscribe to post you created' checked={postCreated} toggle onChange={handlePostCreatedChange} />
+						</Form.Field>
+					</Form.Group>
+					<Form.Group>
```

Comment:
let's comment this out if not working yet from the back-end side.

Reply:
done